web/source/osk/preProcessor.ts

```diff
@@ -48,7 +48,13 @@ namespace com.keyman.osk {
         // Deleting matched deadkeys here seems to correct some of the issues.   (JD 6/6/14)
         outputTarget.deadkeys().deleteMatched();      // Delete any matched deadkeys before continuing
 
-        let keySpec = (e['key'] ? e['key'].spec : null) as keyboards.ActiveKey;
+        // Future note:  we need to refactor osk.OSKKeySpec to instead be a 'tag field' for
+        // keyboards.ActiveKey.  (Prob with generics, allowing the Web-only parts to
+        // be fully specified within the tag.)  
+        //
+        // Would avoid the type shenanigans needed here because of our current type-abuse setup 
+        // for key spec tracking.
+        let keySpec = (e['key'] ? e['key'].spec : null) as unknown as keyboards.ActiveKey;
```

Comment:
Naturally, as is practically tradition for me, I've identified a piece of technical debt that could use some future resolution.  Definitely not touching this right now.

It'd be a good fit with any major OSK refactors that may occur in the future.